pageserver/src/consumption_metrics.rs

```diff
@@ -374,45 +378,31 @@ async fn collect_metrics_iteration(
             },
         }));
 
-        const MAX_RETRIES: u32 = 3;
-
-        for attempt in 0..MAX_RETRIES {
-            let res = client
-                .post(metric_collection_endpoint.clone())
-                .json(&EventChunk {
-                    events: (&chunk_to_send).into(),
-                })
-                .send()
-                .await;
-
-            match res {
-                Ok(res) => {
-                    if res.status().is_success() {
-                        // update cached metrics after they were sent successfully
-                        for (curr_key, curr_val) in chunk.iter() {
-                            cached_metrics.insert(curr_key.clone(), *curr_val);
-                        }
-                    } else {
-                        error!("metrics endpoint refused the sent metrics: {:?}", res);
-                        for metric in chunk_to_send
-                            .iter()
-                            .filter(|metric| metric.value > (1u64 << 40))
-                        {
-                            // Report if the metric value is suspiciously large
-                            error!("potentially abnormal metric value: {:?}", metric);
-                        }
-                    }
-                    break;
-                }
-                Err(err) if err.is_timeout() => {
-                    error!(attempt, "timeout sending metrics, retrying immediately");
-                    continue;
-                }
-                Err(err) => {
-                    error!(attempt, ?err, "failed to send metrics");
-                    break;
-                }
+        use bytes::BufMut;
+
+        // FIXME: this is a new panic we did not have before. not really essential, but panics from
```

Comment:
This comment is good but it would make it better to also explain why we don't think this can fail: we write into a memory buffer.

Reply:
Well you could also have funky types which just fail serialization of some values (have done a few myself :D).

But I forgot I later changed this to be an anyhow::Error... However, I'll fix that in next commits. There is no known serialization problem currently, so this could just go to staging.

Reply:
> Well you could also have funky types which just fail serialization of some values (have done a few myself :D).

Yeah there is a non-zero risk but here the type does not do that. The comment should be there for clarity purposes.

Reply:
I'll move the comment about bytesmut growing to expect...

Reply:
Added in [042cc34](https://github.com/neondatabase/neon/pull/5317/commits/042cc34a51d6a17e0e75a678c35bf32aab50547d)